clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -421,6 +422,17 @@ SYCLToolChain::TranslateArgs(const llvm::opt::DerivedArgList &Args,
   return DAL;
 }
 
+void parseTargetOpts(StringRef ArgString, const llvm::opt::ArgList &Args,
+                     llvm::opt::ArgStringList &CmdArgs) {
+  // Tokenize the string.
+  SmallVector<const char *, 8> TargetArgs;
+  llvm::BumpPtrAllocator A;
+  llvm::StringSaver S(A);
+  llvm::cl::TokenizeGNUCommandLine(ArgString, S, TargetArgs);
```

Comment:
Would this (and making `Args` non-`const`) work too?
```suggestion
  llvm::cl::TokenizeGNUCommandLine(ArgString, S, Args);
```

Reply:
Unfortunately, no.  We need TargetArgs to be a vector specific to the items we are pushing to the backend target compilation.  Using Args would add to the original command line.

Reply:
Ah, of course.